libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.kt

```diff
@@ -215,6 +215,18 @@ open class SalesforceSDKManager protected constructor(
      */
     val loginActivityClass: Class<out Activity> = nativeLoginActivity ?: webviewLoginActivityClass
 
+    /**
+     * For Salesforce Identity API UI Bridge support, the overriding front door bridge URL to
+     * use in place of the default initial login URL
+     */
+    var frontDoorBridgeUrl: String? = null
```

Comment:
When the app receives a deep-linked login QR code from the external QR code reader, that Intent has to be received by the app's main activity such that the login broadcast intent receivers will be registered on a cold start of the app.  In that event, the main application will be about to trigger login since the call to fetch the REST client will automatically display login as a side effect and return null.  These two new variables our on SDK manager singleton give the client manager and login activity the hint that there is a pending front door URL to use.

Overall, I feel we'd be better if the existing SDK design gave us another means of doing this.  But, there are a couple patterns in place we'd have to resolve first for trust work:
- Either move the broadcast intent receivers to the application class rather than a specific activity or publish login events on another bus
- Make the display of the login activity pass through the app rather than be a side effect of getting the REST client
- Remove the singleton pattern of the SDK manager

I've been thinking about a more future-forward means of passing these two data points between the deep-link activity (main application) and the login activity in the context of MSDK's current implementation.  I haven't gotten a good idea yet.  Shall we keep that for a to-do when we clean up the trust work above or is there a cleaner way we could use now?

Reply:
+1 having a TODO for future cleanup (or maybe a story on GUS).